### ISSUE TITLE
ScanObjectNN (paper table 2)

### ISSUE THREAD

**Issue by sheshap** [ajhamdi/MVTN#5]
Hi Thank you so much for the code release.

Can you please give the exact training and evaluation commands used for training and testing the ScanObjectNN dataset to recreate the results of table 2 in paper?

Thanks in advance. Much appreciated.

**Comment by ajhamdi** [comment]
Hi @sheshap 
```python
python run_mvtn.py --data_dir data/ScanObjectNN/ --run_mode train --mvnetwork viewgcn --nb_views 12 --views_config learned_spherical --pc_rendering --pretrained --shape_extractor PointNet --features_type logits --clip_grads  
```

**Comment by sheshap** [comment]
@ajhamdi 

Few of the arguments mentioned in the above command are not recognized.

run_mvtn.py: error: unrecognized arguments: --pretrained --shape_extractor PointNet --features_type logits --clip_grads

**Comment by ajhamdi** [comment]
Because the `config.yaml` configuration file has them by default already. This command should do 

```bash
python run_mvtn.py --data_dir data/ScanObjectNN/ --run_mode train --mvnetwork viewgcn --nb_views 12 --pc_rendering
```

**Comment by sheshap** [comment]
@ajhamdi 

The above command resulted in the best training accuracy of 89.02% and test accuracy of 87.5 on the obj_only variant.

**Comment by ajhamdi** [comment]
Are you using one-stage training or two-stage? The results in the paper reported are for two-stage in which the first stage is trained on ModelNet ( CNN ) . Also, make sure you are setting `canonical_distance:  1.0` in the `config.yaml` file

**Comment by sheshap** [comment]
Please confirm if the following two commands represent the two stages. Much appreciated.

The first stage is 50 epochs of training the backbone CNN on the single view images 

1)  python run_mvtn.py --data_dir data/ModelNet40/ --run_mode train --mvnetwork mvcnn --nb_views 1 --views_config learned_spherical --pc_rendering 

The second stage is 35 epochs on the multi-view network on the M views of the 3D object.

2)  python run_mvtn.py --data_dir data/ScanObjectNN/ --run_mode train --mvnetwork viewgcn --nb_views 12 --views_config learned_spherical --pc_rendering --resume_second true --resume_first false


Thanks

**Comment by sheshap** [comment]
@ajhamdi Can you please help with exact configurations/commands to recreate results on ScanObjectNN?

**Comment by sheshap** [comment]
@ajhamdi 

I used the below command and got 82.5% on PB_T50_RS

python run_mvtn.py --data_dir data/ScanObjectNN --run_mode train --mvnetwork mvcnn --nb_views 12 --views_config learned_spherical --pc_rendering -dsetp hardest

Can you please provide the exact configurations/commands to recreate results on ScanObjectNN based on two-phase training for viewgcn?

Thanks in Advance.

**Comment by sheshap** [comment]
I was able to reproduce 92.6% (scanobjectnn - with_bg) using the below commands.

python run_mvtn.py --data_dir ../MVTN/data/ModelNet40/ --run_mode train --mvnetwork viewgcn --nb_views 1 --views_config learned_spherical --pc_rendering --viewgcn_phase first

python run_mvtn.py --data_dir data/ScanObjectNN/ --run_mode train --mvnetwork viewgcn --nb_views 12 --views_config learned_spherical  --pc_rendering --viewgcn_phase  second --dset_variant with_bg

Closing the issue. Thanks
